packages/fx-core/tests/plugins/resource/appstudio/unit/createManifestForRemote.test.ts

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import "mocha";
+import * as chai from "chai";
+import { AppStudioPlugin } from "./../../../../../src/plugins/resource/appstudio";
+import { AppStudioPluginImpl } from "./../../../../../src/plugins/resource/appstudio/plugin";
+import { TeamsBot } from "./../../../../../src/plugins/resource/bot";
+import { AppStudioError } from "./../../../../../src/plugins/resource/appstudio/errors";
+import {
+  ConfigMap,
+  PluginContext,
+  TeamsAppManifest,
+  ok,
+  err,
+  Plugin,
+} from "@microsoft/teamsfx-api";
+import * as uuid from "uuid";
+import fs from "fs-extra";
+import sinon from "sinon";
+import { AppStudioResultFactory } from "../../../../../src/plugins/resource/appstudio/results";
+
+describe("Reload Manifest and Check Required Fields", () => {
+  let plugin: AppStudioPlugin;
+  let internalError_ctx: PluginContext;
+  let ctx: PluginContext;
+  let manifest: TeamsAppManifest;
+  let BotPlugin: Plugin;
+  let selectedPlugins: Plugin[];
+  const sandbox = sinon.createSandbox();
+
+  beforeEach(async () => {
+    plugin = new AppStudioPlugin();
+    internalError_ctx = {
+      root: "./",
+      configOfOtherPlugins: new Map(),
+      config: new ConfigMap(),
+      app: new TeamsAppManifest(),
+    };
+    internalError_ctx.projectSettings = {
+      appName: "my app",
+      currentEnv: "default",
+      projectId: uuid.v4(),
+      solutionSettings: {
+        name: "azure",
+        version: "1.0",
+      },
+    };
+
+    ctx = {
+      root: "./",
+      configOfOtherPlugins: new Map(),
+      config: new ConfigMap(),
+      app: new TeamsAppManifest(),
+    };
+    ctx.projectSettings = {
+      appName: "my app",
+      currentEnv: "default",
+      projectId: uuid.v4(),
+      solutionSettings: {
+        name: "azure",
+        version: "1.0",
+        capabilities: ["Bot"],
+      },
+    };
+    manifest = new TeamsAppManifest();
+    const botplugin: Plugin = new TeamsBot();
+    BotPlugin = botplugin as Plugin;
+    BotPlugin.name = "fx-resource-bot";
+    BotPlugin.displayName = "Bot";
+    selectedPlugins = [BotPlugin];
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it("maybeSelectedPlugins error", async () => {
```

Comment:
BDD style tests are expected to be written in a way that a test can be read out. 
Something like it("should return error")

Reply:
Thanks for the reminder!  I have already change the testcase names.